onboard/merge_iterator_test.go

```diff
@@ -0,0 +1,91 @@
+package onboard
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/treeverse/lakefs/catalog/rocks/testutils"
+
+	"github.com/treeverse/lakefs/catalog/rocks"
+)
+
+func TestMergeIterator(t *testing.T) {
+	cases := []struct {
+		Name        string
+		InvIt       rocks.EntryIterator
+		CommittedIt rocks.EntryListingIterator
+		Prefixes    []string
+		Results     []string
+	}{
+		{
+			Name: "Simple prefix",
+			InvIt: testutils.NewFakeEntryIterator([]*rocks.EntryRecord{
+				{Path: "a/b/c"},
+				{Path: "a/d/c"},
+			}),
+			CommittedIt: rocks.NewEntryListingIterator(testutils.NewFakeEntryIterator(
+				[]*rocks.EntryRecord{
+					{Path: "a/b/d"},
+					{Path: "a/b/c/d"},
+					{Path: "a/c/d"},
+					{Path: "b/c/d"},
+				}), "", ""),
+			Prefixes: []string{"a/"},
+			Results:  []string{"a/b/c", "a/d/c", "b/c/d"},
+		},
+		{
+			Name: "Empty prefix - All inventory",
+			InvIt: testutils.NewFakeEntryIterator([]*rocks.EntryRecord{
+				{Path: "a/b/c"},
+				{Path: "e/f/s/a"},
+			}),
+			CommittedIt: rocks.NewEntryListingIterator(testutils.NewFakeEntryIterator(
+				[]*rocks.EntryRecord{
+					{Path: "some/other/prefix"},
+					{Path: "some/other/prefix/c"},
+					{Path: "some/other/prefix/e"},
+				}), "", "/"),
```

Comment:
Is delimiter non-empty here intentionally?  If so, please reflect that in the test name.

Reply:
Not intentional, nice catch!